mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/datadriven/JsonDriven.java

```diff
@@ -33,19 +35,34 @@ private JsonDriven() {
 				.fromJson(fileToJson(filename), clazz);
 	}
 	
+	public static <T> T provide(Reader reader, Class<T> clazz) {
+		return new GsonBuilder().create()
+				.fromJson(readerToJson(reader), clazz);
+	}
+	
 	/**
 	 * @param fileName
+	 *            fileName
 	 * @return json from file or empty object when parsing file failed
 	 */
 	private static JsonArray fileToJson(String fileName) {
-		JsonArray jsonArray = new JsonArray();
-		try {
-			jsonArray = new JsonParser().parse(new FileReader(fileName))
-					.getAsJsonArray();
+		try (Reader reader = new FileReader(fileName)) {
+			return readerToJson(reader);
 		} catch (FileNotFoundException e) {
 			BFLogger.logError("Json file not found: " + fileName);
+		} catch (IOException e) {
+			BFLogger.logError("Json file could not be read: " + fileName);
 		}
-		return jsonArray;
+		return new JsonArray();
```

Comment:
Why we are parsing Json again, when it is parsed at this place and we could return this?